### ISSUE TITLE
Parentless tasks try to spawn again after already finishing

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-flow#5002]
#### Describe the bug

(Back compat mode, possible follow-up to #4906)

A failed, parentless task that was suicided tries to spawn again.

#### Steps to reproduce the bug
Take the following `suite.rc` running in back-compat mode.
```ini
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    [[dependencies]]
        [[[P1]]]
            graph = """
                foo:fail => recover
                foo | recover => bar

                foo => !recover
                recover => !foo
            """
[runtime]
    [[foo]]
        script = false
    [[bar, recover]]
        script = true
```

You get a message that a `foo` suicides after `recover` succeeds. However, a bit further down you get "foo already spawned in {1}"

```
CRITICAL - [2/foo failed job:01 flows:1] task proxy removed (suicide)
INFO - [3/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:37+01:00
INFO - [3/recover running job:01 flows:1] => succeeded
INFO - [3/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [3/foo failed job:01 flows:1] task proxy removed (suicide)
INFO - [4/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:37+01:00
INFO - [4/recover running job:01 flows:1] => succeeded
INFO - [4/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [4/foo failed job:01 flows:1] task proxy removed (suicide)
INFO - [5/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:37+01:00
INFO - [5/recover running job:01 flows:1] => succeeded
INFO - [5/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [5/foo failed job:01 flows:1] task proxy removed (suicide)
...
INFO - [1/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:38+01:00
INFO - [1/recover running job:01 flows:1] => succeeded
WARNING - Task 2/foo already spawned in {1}
WARNING - Task 3/foo already spawned in {1}
WARNING - Task 4/foo already spawned in {1}
WARNING - Task 5/foo already spawned in {1}
INFO - [1/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [1/foo failed job:01 flows:1] task proxy removed (suicide)
```

#### Additional context

(Update: I think this next bit might have been fixed by #4906 actually)

In some circumstances you might get `foo failed -> waiting` before the suicide message and then `foo` submits a second time. This might be why I had the mysterious long-lived "preparing" jobs in #4994. This is hard to reproduce ~(may be a random combination of pausing/resuming/holding/releasing/stopping/restarting that did it)~ (probably fixed by #4906).

```
INFO - [3/recover running job:01 flows:1] => succeeded
INFO - [3/foo failed job:01 flows:1] => waiting
INFO - [3/foo waiting job:01 flows:1] => waiting(queued)
CRITICAL - [3/foo waiting(queued) job:01 flows:1] task proxy removed (suicide)
...
INFO - [3/foo waiting job:01 flows:1] => preparing
...
INFO - [3/foo preparing job:02 flows:1] host=localhost
```

**Comment by hjoliver** [comment]
@MetRonnie - I've been playing with your example and I can't seem to reproduce the problem.
- prior to merge of 4906 I see `WARNING - Task n/recover already spawned in {1}` (i.e. `recover`, not `foo`)
- on master, I see no problem at all

**Comment by wxtim** [comment]
I _have_ managed to reproduce this, although not for every cycle - and then sometimes multiple times for the same cycle.

**Comment by wxtim** [comment]
I thought I could reproduce it, then I updated master...

**Comment by wxtim** [comment]
I'm able to reproduce this, although it doesn't appear to happen very often:
```
Every 0.7s: grep "already spawned in " temp/runN/log/scheduler/log                                                                                                                         Fri Jul 22 10:00:36 2022

2022-07-22T09:58:32+01:00 WARNING - Task 25/foo already spawned in {1}
2022-07-22T09:59:53+01:00 WARNING - Task 32/foo already spawned in {1}

```

**Comment by MetRonnie** [comment]
It's true, however I don't think it should be attempting to spawn. It seems to be trying to spawn `foo` off the back of `recover`

**Comment by hjoliver** [comment]
A-ha, OK I have reproduced it. Turns out I had my process pool size strapped down to 1, after testing some other problem, and that put the brakes on the workflow enough to prevent this happening.

I suspect it's not actually a problem though, just taking a look...

**Comment by hjoliver** [comment]
```console
$ cylc cat-log bug | grep WARNING
2022-07-22T23:14:17+12:00 WARNING - Backward compatibility mode ON
2022-07-22T23:14:28+12:00 WARNING - Task 4/foo already spawned in {1}
2022-07-22T23:14:28+12:00 WARNING - Task 5/foo already spawned in {1}
2022-07-22T23:14:45+12:00 WARNING - Task 7/foo already spawned in {1}
2022-07-22T23:14:45+12:00 WARNING - Task 8/foo already spawned in {1}
2022-07-22T23:14:45+12:00 WARNING - Task 10/foo already spawned in {1}
2022-07-22T23:14:45+12:00 WARNING - Task 10/foo already spawned in {1}
2022-07-22T23:15:02+12:00 WARNING - Task 13/foo already spawned in {1}
2022-07-22T23:15:19+12:00 WARNING - Task 19/foo already spawned in {1}
2022-07-22T23:16:11+12:00 WARNING - Task 33/foo already spawned in {1}
2022-07-22T23:16:11+12:00 WARNING - Task 34/foo already spawned in {1}
2022-07-22T23:16:11+12:00 WARNING - Task 35/foo already spawned in {1}
2022-07-22T23:16:28+12:00 WARNING - Task 39/foo already spawned in {1}
2022-07-22T23:16:45+12:00 WARNING - Task 45/foo already spawned in {1}
2022-07-22T23:16:45+12:00 WARNING - Task 45/foo already spawned in {1}
2022-07-22T23:16:45+12:00 WARNING - Task 45/foo already spawned in {1}
2022-07-22T23:17:02+12:00 WARNING - Task 47/foo already spawned in {1}
2022-07-22T23:17:02+12:00 WARNING - Task 48/foo already spawned in {1}
2022-07-22T23:17:02+12:00 WARNING - Task 49/foo already spawned in {1}
```

**Comment by MetRonnie** [comment]
Maybe more of a https://github.com/cylc/cylc-flow/labels/could%20be%20better?

**Comment by hjoliver** [comment]
I fixed it for 8.0. Any bug in the scheduling logic scares me.
